### PR TITLE
Update CESQL tck tests to match spec changes

### PR DESCRIPTION
**Pull request by Cali0707**
With all of the CESQL spec changes recently, the spec needed some updates to reflect what had changed

## Proposed Changes

- Add results to every expression that has an error, as CESQL spec has been clarified on what should happen in that case
- Added missing attribute errors back, as well as tests to ensure that operators handle it correctly
- Add test for the `ABS` function overflow error
- Fix boolean casting now that it is defined for all types

### CODE REVIEW

File: cesql/cesql_tck/README.md [hunk 1]
```diff
@@ -25,3 +25,4 @@ The `error` values could be any of the following:
 * `cast`: Casting error
 * `missingFunction`: Addressed a missing function
 * `functionEvaluation`: Error while evaluating a function
+* `missingAttribute`: Error due to a missing attribute
```
Comment (duglin) [inline cesql/cesql_tck/README.md:28]: I think you added a generic error message to the spec - should we test for that too? It might end up being more of an impl-specific error case. If it's too hard to put into this generic tck stuff then I'm ok with skipping it. Was just curious.

File: cesql/cesql_tck/negate_operator.yaml [hunk 2]
```diff
@@ -14,7 +14,11 @@ tests:
     expression: --'10'
     result: 10
 
-  - name: Invalid boolean cast
+  - name: Minus with boolean cast
     expression: -TRUE
+    result: -1
```
Comment (duglin) [inline cesql/cesql_tck/negate_operator.yaml:19]: Correct result, but funny

File: cesql/cesql_tck/negate_operator.yaml [hunk 3]
```diff
@@ -14,7 +14,11 @@ tests:
...
+
+  - name: Minus with missing attribute
+    expression: -missing
     result: 0
-    error: cast
+    error: match
```
Comment (duglin) [inline cesql/cesql_tck/negate_operator.yaml:24]: What does "match' mean here - I don't see it listed as a valid error? I expected "missingAttribute"

**Comment by duglin** [comment]
Just a few minor questions... otherwise LGTM

File: cesql/cesql_tck/negate_operator.yaml (hunk 3 above)
Comment (Cali0707) [reply cesql/cesql_tck/negate_operator.yaml:24]: Tbh I'm not sure how I ended up with `match` here...

You are totally correct, it should be `missingAttribute`

File: cesql/cesql_tck/README.md (hunk 1 above)
Comment (Cali0707) [reply cesql/cesql_tck/README.md:28]: I think we could mention it in the `README` for completeness, but I can't think of any use cases where we would actually use it

**Comment by duglin** [comment]
Approved on the 6/6 call
